src/common/dpycmn.cpp

```diff
@@ -216,33 +242,86 @@ bool wxDisplay::ChangeMode(const wxVideoMode& mode)
 // wxDisplayFactory implementation
 // ============================================================================
 
+bool wxDisplayFactory::UpdateDisplayChange(wxDisplayImpl &impl) const
+{
+    // If the factory is not able to detect connection state of the
+    // input display we assume that it is disconnected now.
+    impl.Disconnect();
+    return false;
+}
+
 void wxDisplayFactory::ClearImpls()
 {
+    m_impls.clear();
+}
+
+void wxDisplayFactory::UpdateDisplayChanges()
+{
+    m_countOnLastAccess = GetCount();
+    wxVector<wxObjectDataPtr<wxDisplayImpl> > impls;
+    impls.reserve(m_countOnLastAccess);
+
     for ( size_t n = 0; n < m_impls.size(); ++n )
     {
-        // It can be null, that's ok.
-        delete m_impls[n];
+        wxObjectDataPtr<wxDisplayImpl> &impl = m_impls[n];
+
+        // Try to update display state or mark it as disconnected.
+        if ( UpdateDisplayChange(*impl) )
```

Comment:
To continue my other comment, this should basically just become

```suggestion
        if ( impl->UpdateOnDisplayChange() )
```

Reply:
Function name updated to `UpdateOnDisplayChange` in cdc7987